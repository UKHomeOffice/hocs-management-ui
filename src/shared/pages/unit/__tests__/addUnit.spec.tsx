import React from 'react';
import { match } from 'react-router';
import { createBrowserHistory, History, Location } from 'history';
import { act, render, RenderResult, wait } from '@testing-library/react';
import AddUnit from '../addUnit';
import Unit from '../../../models/unit';

let match: match<any>;
let history: History<any>;
let location: Location;
let mockState: Unit;

const useReducerSpy = jest.spyOn(React, 'useReducer');
const dispatch = jest.fn();

beforeEach(() => {
    history = createBrowserHistory();
    match = {
        isExact: true,
        params: { teamId: '__teamId__' },
        path: '',
        url: ''
    };

    location = {
        hash: '',
        key: '',
        pathname: '',
        search: '',
        state: {}
    };
    mockState = {
        displayName: '',
        shortCode: ''
    };
    useReducerSpy.mockImplementationOnce(() => [mockState, dispatch]);
    dispatch.mockReset();
});

describe('when the addToTeam component is mounted', () => {
    it('should render with default props', async () => {
        expect.assertions(1);
        let wrapper: RenderResult;
        act(() => {
            wrapper = render(<AddUnit history={history} location={location} match={match}></AddUnit>);
        });

        await wait(() => {
            expect(wrapper.container).toMatchSnapshot();
        });
    });
});
