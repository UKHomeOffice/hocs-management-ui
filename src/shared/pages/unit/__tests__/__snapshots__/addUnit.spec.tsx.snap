// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the addToTeam component is mounted should render with default props 1`] = `
<div>
  <div>
    <form
      action="/api/units"
      method="POST"
    >
      <input
        name="_csrf"
        type="hidden"
        value=""
      />
      <div
        class="govuk-form-group"
      >
        <label
          class="govuk-label govuk-label--s"
          for="displayName"
          id="displayName-label"
        >
          Display Name
        </label>
        <input
          class="govuk-input"
          id="displayName"
          name="displayName"
          type="text"
          value=""
        />
      </div>
      <div
        class="govuk-form-group"
      >
        <label
          class="govuk-label govuk-label--s"
          for="shortCode"
          id="shortCode-label"
        >
          Short Code
        </label>
        <input
          class="govuk-input"
          id="shortCode"
          name="shortCode"
          type="text"
          value=""
        />
      </div>
      <input
        class="govuk-button"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
